novawallet/Common/RampProvider/RampProviderProtocol.swift

```diff
@@ -87,20 +88,31 @@ protocol OffRampHookFactoryProvider {
     var offRampHookFactory: OffRampHookFactoryProtocol { get }
 }
 
-extension RampProviderProtocol where Self: BaseURLStringProvider, Self: OffRampHookFactoryProvider {
-    func buildOffRampHooks(
+protocol OnRampHookFactoryProvider {
```

Comment:
I think `OnRampHookFactoryProviding` or `OnRampHookFactoryProviderProtocol` fits better as we usually use nouns like `OnRampHookFactoryProvider` for concrete implementations of a protocol